### PR TITLE
Jonathan Admin Can view any play page

### LINKED ISSUE #5: FEATURE: Admins should be able to visit any play page "as if" they were that user (maybe read only?)
# User Story

* As an admin
* I can see what any players page would look like
* So that I can better understand how the game is playing out.

# Discussion

It would be nice if the admin could "drop in" on the play page of any player in the game, and see their play page "as if" they were that user.

It might be better if this were in some kind of "read only" mode, where they can see all of the data "as if" they were that user, but they can't buy/sell cows on behalf of that user.

But it's ok to implement it so that the admin *can* do those things if that's easier.

# Consider a Spike

Note that this issue is one of those that might be dead simple, or really really complex, and it's difficult to know up front.  You may need to do what's called a "spike" where you do a practice run on trying to build it in a Draft PR that is not intended to be merged, but functions as a kind of "first draft" where you try out ideas.  

In a spike, since it's not intended as code to be merged, you can cut corners; e.g. 
* you can leave in all of the `console.log()` and/or `log.info()` statements you like
* you can have commented out code all over the place
* you don't have to worry about testing or test coverage
* it's ok if the user interface is a little messy

You just try out your plan for implementing the feature to see if your plan is feasible.  If it seems to be feasible, then you make a new branch and a new PR, doing everything "the right way", using the spike as a "cheat sheet".

### Acceptance Criteria

- [x] When an admin views the Leaderboard there is an extra button called "Play Page"
- [x] This button does not show up on the Leaderboard for any user that is not an Admin
- [ ] When the admin clicks that button, they are taken to the url `/admin/play/:userCommonsId`
- [x] What they see is the PlayPage, but they see it "as if" they were the user associated with that userCommonsId.
        This may require some significant refactoring of the PlayPage and how it communicates with the backend!
- [x] The functioning of the PlayPage for regular users is unaffected by these changes.
- [ ] When an admin is viewing the PlayPage "as if" they were that user, there is some visual indication (e.g. a banner at the top of the page) indicating that they are using their admin features to visit another players page. That banner might include: Name of Commons, Name of User, User id, User Email.

### PR DESCRIPTION
**Pull request by jonathan-chen-opus**
## Overview
Dokku Deployment: https://5pm-1-happycows.dokku-01.cs.ucsb.edu/

In this PR, the goal is to set up a page that gives admin account the option to view other players' playpage as if they are players.
The option will be in the leaderboard page, and will be hidden to all regular players.
When visiting the page, the admin will not be able to buy and sell cows, and there will be a banner indicating that this is a admin feature, view only.

## Screenshots (Optional)
On the leader board page, admin can visit a playpage by clicking the link on username.
image

Once enter the page, they can see the number of cows, profits, and other information of the user they are visiting.

https://github.com/ucsb-cs156-f23/proj-happycows-f23-5pm-1/assets/75823989/7b632ca9-942b-4a4a-8992-0c51554e4590

## Future Possibilities 
I think there can be future admin features based on this.
Once enter the play page, maybe there can be more actions to performed.
As a admin, the ability to oversee the whole common and understand how each user is doing is pretty important.

## Tests
- [ ] Backend Unit tests (`mvn test`) pass
- [ ] Backend Test coverage (`mvn test jacoco:report`) 100%
- [ ] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [ ] Frontend Unit tests (`npm test`) pass
- [ ] Frontend Test coverage (`npm run coverage`) 100%
- [ ] Frontend Mutation tests (`npx stryker run`) 100% 
- [ ] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues

Closes #5

### CODE REVIEW

File: frontend/src/tests/components/Commons/ProfitsTable.test 2.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { render, screen, waitFor } from "@testing-library/react";
```
Comment (pconrad) [inline frontend/src/tests/components/Commons/ProfitsTable.test 2.js]: Why is there a ProfitsTable.test 2.js file?

File: frontend/src/tests/pages/AdminViewPlayPage.test 2.js [hunk 2]
```diff
@@ -0,0 +1,284 @@
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
```
Comment (pconrad) [inline frontend/src/tests/pages/AdminViewPlayPage.test 2.js]: Why is there a AdminViewPlayPage.test 2.js?

**Review by pconrad — CHANGES_REQUESTED** [review]
There seem to be a few stray files here and there.

File: frontend/src/tests/pages/AdminViewPlayPage.test.js [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+                    <AdminViewPlayPage />
+                </MemoryRouter>
+            </QueryClientProvider>
+        );
+
+        expect(await screen.findByText(/Announcements/)).toBeInTheDocument();
+        expect(await screen.findByTestId("CommonsPlay")).toBeInTheDocument();
+    });
+
+    // test("Make sure div has correct attributes", async () => {
```
Comment (pconrad) [inline frontend/src/tests/pages/AdminViewPlayPage.test.js]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

**Review by pconrad — CHANGES_REQUESTED** [review]
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

**Review by AngelG261 — DISMISSED** [review]
LGTM!

**Review by pconrad — DISMISSED** [review]
LGTM.. but will want to see a deployment on dokku before merging

**Review by pconrad — APPROVED** [review]
LGTM

I do have some UX concerns, but we can put those in another PR

**Pull request merged**